crates/anvil-polkadot/src/api_server/server.rs

```diff
@@ -1455,6 +1474,59 @@ impl ApiServer {
 
         Ok(())
     }
+
+    async fn wait_for_hash(
+        &self,
+        receiver: Option<tokio::sync::broadcast::Receiver<H256>>,
+        awaited_hash: H256,
+    ) -> Result<()> {
+        if let Some(mut receiver) = receiver {
+            tokio::time::timeout(Duration::from_millis(500), async {
+                loop {
```

Comment:
if you build a large number of blocks this timeout could probably be achieved. Let's add a couple of seconds here to be safe (2 seconds maybe)